code/datums/components/material/remote_materials.dm

```diff
@@ -145,16 +145,27 @@ handles linking back and forth.
 		if(!check_z_level(M.buffer))
 			to_chat(user, span_warning("[parent] is too far away to get a connection signal!"))
 			return COMPONENT_BLOCK_TOOL_ATTACK
+
+		var/obj/machinery/ore_silo/new_silo = M.buffer
+		var/datum/component/material_container/new_container = new_silo.GetComponent(/datum/component/material_container)
 		if (silo)
 			silo.ore_connected_machines -= src
 			silo.holds -= src
 			silo.updateUsrDialog()
 		else if (mat_container)
+			//transfer all mats to silo. whatever cannot be transfered is dumped out as sheets
+			if(mat_container.total_amount())
+				for(var/datum/material/MAT as anything in mat_container.materials)
```

Comment:
i thought MAT was a define at first, use snake case for this please. rest is good though

Reply:
Done